crates/shared/src/gas_price_estimation/fake.rs

```diff
@@ -0,0 +1,21 @@
+use {
+    anyhow::Result,
+    gas_estimation::{GasPrice1559, GasPriceEstimating},
+    std::sync::{Arc, Mutex},
+};
+
+#[derive(Default)]
+pub struct FakeGasPriceEstimator(pub Arc<Mutex<GasPrice1559>>);
```

Comment:
Does this need to be a Mutex?

From my search around the codebase, `FakeGasPriceEstimator` gets wrapped in an `Arc` anyway and never mutated internally, so if I'm not mistaken, `.0` specifically doesn't even need to be `Arc`, it can always be read as `&`